release/scripts/modules/rna_keymap_ui.py

```diff
@@ -122,11 +122,8 @@ def draw_kmi(display_keymaps, kc, km, kmi, layout, level):
 
     col = _indented_layout(layout, level)
 
-    if kmi.show_expanded:
-        col = col.column(align=True)
-        box = col.box()
-    else:
-        box = col.column()
+    col = col.column(align=True)
+    box = col.box()
```

Comment:
I think, this part solved the issue of "Unexpand Upon Change" behavior I have regularly encountered in the past.  Don't know if anyone else had the same issue.

The problem might have been that, depending on this `kmi.show_expanded` value, different types of UI elements are generated.  And somehow also resets this `kmi.show_expanded` value.  This is only one of the assumptions for why the "Unexpand Upon Change" behavior.

By eliminating the logic to create 2 streams of possibility, the problem seems to be fixed.